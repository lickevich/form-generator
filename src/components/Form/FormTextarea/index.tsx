import React from "react";

import { IInput } from "../../../type";
import Row from "../Row";
import Col from "../Col";

const FormTextarea: React.FC<IInput> = ({
  index = "",
  label = "",
  ...rest
}) => {
  const id = `${label}${index}`;

  return (
    <Row>
      <Col>
        <label htmlFor={id} title={label}>
          {label}
        </label>
      </Col>
      <Col>
        <textarea id={id} {...rest} />
      </Col>
    </Row>
  );
};

export default FormTextarea;
