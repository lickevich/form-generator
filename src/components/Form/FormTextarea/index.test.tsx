import React from "react";
import { render } from "@testing-library/react";

import FormTextarea from "./";

describe("FormTextarea", () => {
  it("render the component", () => {
    const component = (
      <FormTextarea index="1" label="Label_textarea" className="active" />
    );
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
