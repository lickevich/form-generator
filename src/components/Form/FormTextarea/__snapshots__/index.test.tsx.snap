// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextarea render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa kTYFsi"
  >
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <label
        for="Label_textarea1"
        title="Label_textarea"
      >
        Label_textarea
      </label>
    </div>
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <textarea
        class="active"
        id="Label_textarea1"
      />
    </div>
  </div>
</DocumentFragment>
`;
